llvm/include/llvm/CodeGen/CodeGenPassBuilder.h

```diff
@@ -134,7 +167,19 @@ template <typename DerivedT> class CodeGenPassBuilder {
   }
 
   PassInstrumentationCallbacks *getPassInstrumentationCallbacks() const {
-    return PIC;
+    static PassInstrumentationCallbacks PseudoPIC;
+    return PIC ? PIC : &PseudoPIC;
```

Comment:
`PassBuilder` doesn't have a `static` `PIC` though. it's ok if it's `nullptr` since that's handled [here](https://github.com/llvm/llvm-project/blob/34a65980d7d2e1b05e3fc88535cafe606ee55e04/llvm/include/llvm/IR/PassInstrumentation.h#L230)

Reply:
similar to the start/stop, I think this would be better handled by not adding the pass at all to the pipeline by checking in `addPass`, rather than working around it later on using pass instrumentation